### PR TITLE
feat(route): add dlnews

### PR DESCRIPTION
**Pull request by Rjnishant530**
## 该 PR 相关 Issue / Involved Issue

Close #

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/dlnews
/dlnews/llama-u
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [ ] 中文文档 CN
  - [x] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [x] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by Rjnishant530** [comment]
this site has rate limit, thats why the feed is failing, pls guide

File: lib/v2/dlnews/category.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    .clone()
+                    .children('div.move-to-right')
+                    .remove()
+                    .end()
+                    .children('div#comments-form-wrapper')
+                    .remove()
+                    .end()
+                    .map((index, element) => $(element).html())
+                    .get()
+                    .join('');
```
Comment (TonyRL) [inline lib/v2/dlnews/category.js:54]: Please use the embedded `Fusion.globalContent` object instead.
docs: https://docs.rsshub.app/en/joinus/new-rss/before-start.html#acquire-data

**Comment by Rjnishant530** [comment]
http://localhost:1200/dlnews/llama-u

File: lib/v2/dlnews/category.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                                    data = '';
+                                }
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+
+                    return acc + data;
+                }, '');
```
Comment (TonyRL) [inline lib/v2/dlnews/category.js:94]: Please use art-template. docs: https://docs.rsshub.app/en/joinus/advanced/script-standard#rendering-templates
The template will look somehow like
```handlebars
{{ each data d }}
    {{ if d.type === 'custom_embed' }}
        <ul>
            {{ each d.embed.config.text.split('\n') line }}
                <li>{{ line }}</li>
            {{ /each }}
        </ul>
    {{ else if d.type === 'oembed_response' }}
        {{@ d.raw_oembed.html }}
    {{ else if d.type === 'header' }}
        <h{{ d.level }}>{{ d.content }}</h{{ d.level }}>
    {{ else if d.type === 'image' }}
        <figure>
            <img src="{{ d.url }}" alt="{{ d.alt_text }}">
            <figcaption>{{ d.subtitle }}</figcaption>
        </figure>
    {{ else if d.type === 'list' }}
        {{ if d.list_type === 'unordered' }}<ul>{{ else }}<ol>{{ /if }}
            {{ each d.items item }}
                <li>{{@ item.content }}</li>
            {{ /each }}
        {{ if d.list_type === 'unordered' }}</ul>{{ else }}</ol>{{ /if }}
    {{ else if d.type === 'text' }}
        <p>{{@ d.content }}</p>
    {{ /if }}
{{ /each }}

```

File: lib/v2/dlnews/category.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    };
+    const data = await getData(`${baseUrl}${apiPath}?query=${encodeURIComponent(JSON.stringify(query))}&_website=${_website}`);
+    const list = getList(data);
+    const items = await Promise.all(
+        list.map((item) =>
+            ctx.cache.tryGet(item.link, async () => {
+                const { data: response } = await got(item.link);
+                const $ = cheerio.load(response);
+                const scriptTagContent = $('script#fusion-metadata').text();
+                const jsonData = JSON.parse(scriptTagContent.split('Fusion.globalContent=')[1].split('Fusion.globalContentConfig')[0].replace(/(;)/g, '')).content_elements;
```
Comment (TonyRL) [inline lib/v2/dlnews/category.js:45]: You can achieve the same thing using regular expression.
```suggestion
                const jsonData = JSON.parse(scriptTagContent.match(/Fusion\.globalContent=({.*?});Fusion\.globalContentConfig/)[1]).content_elements;
```

File: lib/v2/dlnews/category.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        offset: 0,
+        query: '',
+        size: 15,
+        sort: 'display_date:desc',
+        vertical,
+    };
+    const data = await getData(`${baseUrl}${apiPath}?query=${encodeURIComponent(JSON.stringify(query))}&_website=${_website}`);
+    const list = getList(data);
+    const items = [];
+    for await (const data of asyncPool(1, list, (item) => extractArticle(ctx, item))) {
```
Comment (TonyRL) [inline lib/v2/dlnews/category.js:68]: Can you try any values other than 1?

**Pull request merged**